### ISSUE TITLE
Forge plugin prevents launching tests and debugging from the IDE

### ISSUE THREAD

**Issue by mathieucarbou** [Terracotta-OSS/terracotta-platform#673]
Issue following #267 

@saurabhagas @mobasherul @akomakom : this PRs rollbacks the PR that I have merged yesterday fron @akomakom to integrate the maven forge plugin.

**I saw a BIG blocker when developing this evening: we cannot run or debug anymore the Galvan and Angela tests from the IDE: they are just not launching anymore!**

The reason is that the IDE is scanning the pom.xml to determine **how to run the test**.

The IDE knows how to interpret the surefire and failsafe plugins, but doesn't know anything about the forge plugin.

What is expected is that when running or debugging an Angela or Galvan tests, the IDE will look at the properties set to the failsafe plugin, especially system properties, to use them.

Example: when running Galvan tests, we need to have a kit unzipped and a system property needs to be passed to the test like that:

```
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-failsafe-plugin</artifactId>
        <configuration>
          <systemPropertyVariables>
            <kitInstallationPath>${project.build.directory}/platform-kit-${project.version}</kitInstallationPath>
<!--            <serverDebugPortStart>5005</serverDebugPortStart>-->
          </systemPropertyVariables>
        </configuration>
      </plugin>
```

For Angela tests, this is more compex, there are more system properties.

So the IDEs are able to discover these properties and use them when starting / debugging tests.

Sadly, IDEs do not not about forge plugin.

So guys, please merge this PR which will revert the commits until there is a solution that can be found without impacting the development. **Currently, we are unable to run any IT tests or debug any IT tests by directly running them normally from the IDE.**

**Here is a demonstration of the problem in case you do not see what I am talking about ;-)**

=> https://youtu.be/2u7WgF3Aex0

@akomakom : here are some solutions I see:

1. Develop some Intellij / Netbeans plugins that supports forge
2. Make a profile that completely disables forge and switches back to surefire and failsafe
3. Go back to the user of surefire / failsafe, but we add more flexibility to the Azure init scripts so that we can inject the JAVA_HOME and jvm to use for these plugins through properties (but we need a solutiion that also works locally)

**Comment by akomakom** [comment]
I think that you're overthinking this :)
Just duplicate the block in the pom with a comment, it only appears twice in the project.  Execution is already disabled, so it will only have effect on the IDE.

```diff
Index: dynamic-config/testing/system-tests/pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- dynamic-config/testing/system-tests/pom.xml	(revision 00424c1cb9f989c4bdd8d22f07c48176e6d81be9)
+++ dynamic-config/testing/system-tests/pom.xml	(date 1587126588000)
@@ -168,6 +168,24 @@
           </systemPropertyVariables>
         </configuration>
       </plugin>
+      <!-- this block is for IDE compatibility -->
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-failsafe-plugin</artifactId>
+        <configuration>
+          <systemPropertyVariables>
+            <angela.rootDir>${project.build.directory}/angela</angela.rootDir>
+            <angela.skipUninstall>true</angela.skipUninstall>
+            <angela.tsa.fullLogging>true</angela.tsa.fullLogging>
+            <angela.distribution>${project.version}</angela.distribution>
+            <angela.kitInstallationDir>${project.build.directory}/platform-kit-${project.version}</angela.kitInstallationDir>
+            <angela.java.vendor/> <!--empty to allow any vendor-->
+            <angela.java.version>${java.test.version}</angela.java.version>
+            <org.terracotta.voter.topology.fetch.interval>9000</org.terracotta.voter.topology.fetch.interval>
+            <!--<angela.java.opts>-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005</angela.java.opts>-->
+          </systemPropertyVariables>
+        </configuration>
+      </plugin>
     </plugins>
   </build>
 </project>
Index: management/testing/integration-tests/pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- management/testing/integration-tests/pom.xml	(revision 00424c1cb9f989c4bdd8d22f07c48176e6d81be9)
+++ management/testing/integration-tests/pom.xml	(date 1587126681000)
@@ -176,6 +176,17 @@
           </systemPropertyVariables>
         </configuration>
       </plugin>
+      <!-- this block if for IDE compatibility -->
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-failsafe-plugin</artifactId>
+        <configuration>
+          <systemPropertyVariables>
+            <kitInstallationPath>${project.build.directory}/platform-kit-${project.version}</kitInstallationPath>
+            <!--            <serverDebugPortStart>5005</serverDebugPortStart>-->
+          </systemPropertyVariables>
+        </configuration>
+      </plugin>
     </plugins>
   </build>
 </project>
\ No newline at end of file
```

**Comment by akomakom** [comment]
and if these properties are not harmful, they can all be set in the root pom, for both plugins, in one place

**Comment by mathieucarbou** [comment]
Implemented workaround in #674.
